### ISSUE TITLE
File download has failed

### ISSUE THREAD

**Issue by tingfeng-key** [yaroslavche/phptdlib#64]
Hello, I'm trying to use this library. When using BOT to call download_File, I'll prompt: file download has failed. I'm trying to run in cli mode. The result is still the same. I don't know where the cause is. Please

**Comment by yaroslavche** [comment]
Hi. Please show related code and responses log

**Comment by tingfeng-key** [comment]
code:
```php
$api_id =;
    $api_hash = "";
    $host = "127.0.0.1";
    $port = 7891;
    $token = "";
    $file_id = "";
    try {
        \TDApi\LogConfiguration::setLogVerbosityLevel(\TDApi\LogConfiguration::LVL_ERROR);

        $client = new \TDLib\JsonClient();

        $tdlibParams = new \TDApi\TDLibParameters();
        $tdlibParams
            ->setParameter(\TDApi\TDLibParameters::USE_TEST_DC, false)
            ->setParameter(\TDApi\TDLibParameters::DATABASE_DIRECTORY, '/var/tmp/tdlib')
            ->setParameter(\TDApi\TDLibParameters::FILES_DIRECTORY, '/var/tmp/tdlib')
            ->setParameter(\TDApi\TDLibParameters::USE_FILE_DATABASE, false)
            ->setParameter(\TDApi\TDLibParameters::USE_CHAT_INFO_DATABASE, false)
            ->setParameter(\TDApi\TDLibParameters::USE_MESSAGE_DATABASE, false)
            ->setParameter(\TDApi\TDLibParameters::USE_SECRET_CHATS, false)
            ->setParameter(\TDApi\TDLibParameters::API_ID, $api_id)
            ->setParameter(\TDApi\TDLibParameters::API_HASH, $api_hash)
            ->setParameter(\TDApi\TDLibParameters::SYSTEM_LANGUAGE_CODE, 'en')
            ->setParameter(\TDApi\TDLibParameters::DEVICE_MODEL, php_uname('s'))
            ->setParameter(\TDApi\TDLibParameters::SYSTEM_VERSION, php_uname('v'))
            ->setParameter(\TDApi\TDLibParameters::APPLICATION_VERSION, '0.0.10')
            ->setParameter(\TDApi\TDLibParameters::ENABLE_STORAGE_OPTIMIZER, true)
            ->setParameter(\TDApi\TDLibParameters::IGNORE_FILE_NAMES, false);
        $result = $client->setTdlibParameters($tdlibParams);

        $result = $client->setDatabaseEncryptionKey();

        $state = $client->getAuthorizationState();

        $result = $client->query(json_encode([
            '@type' => 'addProxy',
            'server' => $host,
            'port' => $port,
            'enable' => true,
            'type' => [
                "@type" => "proxyTypeSocks5"
            ]]), 10);
        $result = $client->query(json_encode([
            '@type' => 'checkAuthenticationBotToken',
            'token' => $token
        ]), 10);

        $result = $client->query(json_encode([
            '@type' => 'getRemoteFile',
            "remote_file_id" => $file_id,
            "file_type" => [
                "@type" => "fileTypePhoto"
            ]
        ]), 10);

        $resJson = json_decode($result);
        $result = $client->query(json_encode([
            '@type' => 'downloadFile',
            'file_id' => $resJson->id,
            "priority" => 32,
            "offset" => 0,
            "limit" => 0,
            "synchronous" => true
        ]), 10);
        var_dump($result);
        $allNotifications = $client->getReceivedResponses();
        foreach ($allNotifications as $allNotification) {
            echo $allNotification."<br />";
        }
    } catch (\Exception $exception) {
        echo sprintf('something goes wrong: %s', $exception->getMessage());
    }
```
print result:
```
string(102) "{"@type":"error","code":400,"message":"File download has failed or was cancelled","@extra":1642374265}" {"@type":"updateOption","name":"version","value":{"@type":"optionValueString","value":"1.6.2"}}
{"@type":"updateAuthorizationState","authorization_state":{"@type":"authorizationStateWaitTdlibParameters"}}
{"@type":"updateAuthorizationState","authorization_state":{"@type":"authorizationStateWaitEncryptionKey","is_encrypted":true}}
{"@type":"ok","@extra":2067536153}
{"@type":"updateOption","name":"unix_time","value":{"@type":"optionValueInteger","value":1589082391}}
{"@type":"updateOption","name":"message_text_length_max","value":{"@type":"optionValueInteger","value":4096}}
{"@type":"updateOption","name":"is_location_visible","value":{"@type":"optionValueBoolean","value":false}}
{"@type":"updateOption","name":"forwarded_message_count_max","value":{"@type":"optionValueInteger","value":100}}
{"@type":"updateOption","name":"message_caption_length_max","value":{"@type":"optionValueInteger","value":1024}}
{"@type":"updateOption","name":"pinned_chat_count_max","value":{"@type":"optionValueInteger","value":5}}
{"@type":"updateOption","name":"animated_emoji_sticker_set_name","value":{"@type":"optionValueString","value":"AnimatedEmojies"}}
{"@type":"updateOption","name":"enabled_proxy_id","value":{"@type":"optionValueInteger","value":2}}
{"@type":"updateOption","name":"calls_enabled","value":{"@type":"optionValueBoolean","value":true}}
{"@type":"updateOption","name":"favorite_stickers_limit","value":{"@type":"optionValueInteger","value":5}}
{"@type":"updateOption","name":"t_me_url","value":{"@type":"optionValueString","value":"https://t.me/"}}
{"@type":"updateOption","name":"test_mode","value":{"@type":"optionValueBoolean","value":false}}
{"@type":"updateOption","name":"basic_group_size_max","value":{"@type":"optionValueInteger","value":200}}
{"@type":"updateOption","name":"my_id","value":{"@type":"optionValueInteger","value":888625306}}
{"@type":"updateOption","name":"animated_dice_sticker_set_name","value":{"@type":"optionValueString","value":"AnimatedDices"}}
{"@type":"updateOption","name":"call_connect_timeout_ms","value":{"@type":"optionValueInteger","value":30000}}
{"@type":"updateOption","name":"expect_blocking","value":{"@type":"optionValueBoolean","value":false}}
{"@type":"updateOption","name":"venue_search_bot_username","value":{"@type":"optionValueString","value":"foursquare"}}
{"@type":"updateOption","name":"photo_search_bot_username","value":{"@type":"optionValueString","value":"bing"}}
{"@type":"updateOption","name":"supergroup_size_max","value":{"@type":"optionValueInteger","value":200000}}
{"@type":"updateOption","name":"animation_search_bot_username","value":{"@type":"optionValueString","value":"gif"}}
{"@type":"updateOption","name":"pinned_archived_chat_count_max","value":{"@type":"optionValueInteger","value":100}}
{"@type":"updateOption","name":"call_packet_timeout_ms","value":{"@type":"optionValueInteger","value":10000}}
{"@type":"updateOption","name":"authorization_date","value":{"@type":"optionValueInteger","value":1589018020}}
{"@type":"updateAuthorizationState","authorization_state":{"@type":"authorizationStateReady"}}
{"@type":"ok","@extra":647231364}
{"@type":"updateSelectedBackground","for_dark_theme":false}
{"@type":"updateSelectedBackground","for_dark_theme":true}
{"@type":"updateConnectionState","state":{"@type":"connectionStateConnecting"}}
{"@type":"updateConnectionState","state":{"@type":"connectionStateConnectingToProxy"}}
{"@type":"updateConnectionState","state":{"@type":"connectionStateConnecting"}}
{"@type":"authorizationStateReady","@extra":1012862693}
{"@type":"proxy","id":2,"server":"127.0.0.1","port":7891,"last_used_date":1589082271,"is_enabled":true,"type":{"@type":"proxyTypeSocks5","username":"","password":""},"@extra":80411482}
{"@type":"ok","@extra":708564011}
{"@type":"file","id":1,"size":0,"expected_size":0,"local":{"@type":"localFile","path":"","can_be_downloaded":true,"can_be_deleted":false,"is_downloading_active":false,"is_downloading_completed":false,"download_offset":0,"downloaded_prefix_size":0,"downloaded_size":0},"remote":{"@type":"remoteFile","id":"AgADBQADuqkxG0LzuVXfi3KnK32V0jn2G2t0AAMBAAMCAAN4AANtRwACGAQ","unique_id":"AQADOfYba3QAA21HAAI","is_uploading_active":false,"is_uploading_completed":true,"uploaded_size":0},"@extra":1038510262}
{"@type":"updateFile","file":{"@type":"file","id":1,"size":0,"expected_size":0,"local":{"@type":"localFile","path":"","can_be_downloaded":true,"can_be_deleted":false,"is_downloading_active":true,"is_downloading_completed":false,"download_offset":0,"downloaded_prefix_size":0,"downloaded_size":0},"remote":{"@type":"remoteFile","id":"AgADBQADuqkxG0LzuVXfi3KnK32V0jn2G2t0AAMBAAMCAAN4AANtRwACGAQ","unique_id":"AQADOfYba3QAA21HAAI","is_uploading_active":false,"is_uploading_completed":true,"uploaded_size":0}}}
{"@type":"error","code":400,"message":"File download has failed or was cancelled","@extra":1642374265}
{"@type":"updateFile","file":{"@type":"file","id":1,"size":0,"expected_size":0,"local":{"@type":"localFile","path":"","can_be_downloaded":true,"can_be_deleted":false,"is_downloading_active":false,"is_downloading_completed":false,"download_offset":0,"downloaded_prefix_size":0,"downloaded_size":0},"remote":{"@type":"remoteFile","id":"AgADBQADuqkxG0LzuVXfi3KnK32V0jn2G2t0AAMBAAMCAAN4AANtRwACGAQ","unique_id":"AQADOfYba3QAA21HAAI","is_uploading_active":false,"is_uploading_completed":true,"uploaded_size":0}}}
```

**Comment by yaroslavche** [comment]
Did you try:
 - `->setParameter(\TDApi\TDLibParameters::USE_FILE_DATABASE, true)`
 - `synchronous = false`
?
Actually this is question related to tdlib. I can't anwer why this happens, because I don't know =)

**Comment by tingfeng-key** [comment]
I have solved my problem, thank you, [#1039](https://github.com/tdlib/td/issues/1039)
